lib/opensusebasetest.pm

```diff
@@ -569,6 +569,20 @@ sub wait_boot {
 
             assert_screen 'displaymanager-password-prompt', no_wait => 1;
             type_password $password. "\n";
+            # Sometimes the screen back to displaymanager, so we try to retype password, see poo#45236
+            my $count = 0;
+            while ($count < 3) {
+                if (check_screen('displaymanager', 5) && check_var('DESKTOP', 'gnome')) {
+                    # In GNOME/gdm, we do not have to enter a username, but we have to select it
+                    if (is_tumbleweed) {
+                        send_key 'tab';
+                    }
+                    send_key 'ret';
+                    assert_screen 'displaymanager-password-prompt', no_wait => 1;
+                    type_password $password. "\n";
+                }
+                $count++;
```

Comment:
Missing break the while loop if get the expected result.

Reply:
Need balance whether add break or not.
If not add break,  the function will try catch up gdm page three times and retry pass.
If add break, you have one try when encounter gdm page.
I prefer not add break.